header-rewrite-filter/header_processor_test.cc

```diff
@@ -0,0 +1,52 @@
+#include "gtest/gtest.h"
+#include "header_processor.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace HeaderRewriteFilter {
+
+class ProcessorTest : public ::testing::Test {
+protected:
+    void SetUp() override { }
+};
+
+// Note: both processors assume that the first two arguments are validated (these arguments are validated by the filter)
+
+TEST_F(ProcessorTest, SetHeaderProcessorTest) {
+    SetHeaderProcessor set_header_processor = SetHeaderProcessor();
+
+    std::vector<absl::string_view> operation_expression({"http-request", "set-header", "mock_header", "mock_value"});
+    absl::Status status = set_header_processor.parseOperation(operation_expression);
+    EXPECT_TRUE(status == absl::OkStatus());
+
+    // can set multiple values
+    // TODO: will be changed once append-header operation is added
+    operation_expression = {"http-response", "set-header", "mock_key", "mock_val1", "mock_val2"};
+    status = set_header_processor.parseOperation(operation_expression);
```

Comment:
I think we should add tests for `executeOperation` as well to confirm that the processor is mutating the header map correctly